File: app/src/main/java/com/example/ekycdemo2/model/IDCard.java
package com.example.ekycdemo2.model;

import com.google.firebase.database.Exclude;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class IDCard {

    private String id;
    private String name;
    private String dob;
    private String address;
    private String issuedDate;
    private String issuedAddress;
    @Exclude
    private ArrayList<File> storedFiles = new ArrayList<>(2);
    @Exclude
    public static int FRONT = 0;
    @Exclude
    public static int BACK = 1;

    public IDCard() {

    }

    public IDCard(String id, String name, String dob, String address, String issuedDate, String issuedAddress) {
        this.id = id;
        this.name = name;
        this.dob = dob;
        this.address = address;
        this.issuedDate = issuedDate;
        this.issuedAddress = issuedAddress;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDob() {
        return dob;
    }

    public void setDob(String dob) {
        this.dob = dob;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getIssuedDate() {
        return issuedDate;
    }

    public void setIssuedDate(String issuedDate) {
        this.issuedDate = issuedDate;
    }

    public String getIssuedAddress() {
        return issuedAddress;
    }

    public void setIssuedAddress(String issuedAddress) {
        this.issuedAddress = issuedAddress;
    }

    @Exclude
    public ArrayList<File> getStoredFiles() {
        return storedFiles;
    }

    public void setStoredFiles(ArrayList<File> storedFiles) {
        this.storedFiles = storedFiles;
    }

    public void extract(List<Prediction> predictions) {
        if (id == null) {
            id = predictions.get(0).getOcrText();
            name = predictions.get(1).getOcrText();
            dob = predictions.get(2).getOcrText();
            address = predictions.get(3).getOcrText();
        } else {
            issuedDate = predictions.get(0).getOcrText();
            issuedAddress = predictions.get(1).getOcrText();
        }
    }

    public boolean isFilled() {
        return (id != null) && (issuedDate != null);
    }

}

File: app/src/main/java/com/example/ekycdemo2/model/IDCard.kt
package com.example.ekycdemo2.model

import com.google.firebase.database.Exclude
import com.google.gson.annotations.SerializedName
import java.io.File
import java.util.*


class IDCard {

    var id: String? = null

    @SerializedName("fullname")
    var name: String? = null

    @SerializedName("dob")
    var birthDay: String? = null

    @SerializedName("add")
    var address: String? = null

    var signedAdd: String? = null
    var feature: String? = null

    @SerializedName("date")
    var issuedDay: String? = null

    @SerializedName("month")
    var issueMonth: String? = null;

    @SerializedName("year")
    var issueYear: String? = null;

    @SerializedName("identity_image")
    private val identityImage: String? = null

    @SerializedName("embeddings")

    private val embeddings: List<Int>? = null

    var issuedAdd: String? = null

    @get:Exclude
    var storedFiles = ArrayList<File>(2)

    override fun toString(): String {
        return "IDCard {" +
                "\n ID= " + id +
                "\n Tên= '" + name + '\'' +
                "\n Ngày sinh= " + birthDay +
                "\n Nguyên quán= '" + address + '\'' +
                "\n Nơi ĐKHK thường trú= '" + address + '\'' +
                "\n Ngày đăng ký= '" + issuedDay + "/" + issueMonth + "/" + issueYear + '\'' +
                "\n}"
    }

    companion object {
        const val FRONT = 0;
        const val BACK = 1;
    }

    fun extract(idCard: IDCard) {
        if (!idCard.id.isNullOrEmpty()) {
            id = idCard.id;
            name = idCard.name;
            birthDay = idCard.birthDay;
            address = idCard.address;
        } else {
            issuedDay = idCard.issuedDay;
            issueMonth = idCard.issueMonth;
            issueYear = idCard.issueYear;
        }
    }

    var issuedDate = "$issuedDay/$issueMonth/$issueYear";
}